### PR TITLE
MISUV-8364 Add Advanced Rate to Scottish Tax Calc

### PR DESCRIPTION
**Pull request by tioadesola1**
**Waiting on the welsh for the ticket.** But view been updated to account for new tax band.

### CODE REVIEW

File: app/connectors/IncomeTaxCalculationConnector.scala [hunk 1]
```diff
@@ -48,7 +48,10 @@ class IncomeTaxCalculationConnector @Inject()(http: HttpClient,
                 s"Json validation error parsing calculation response, error $invalid")
               LiabilityCalculationError(INTERNAL_SERVER_ERROR, "Json validation error parsing calculation response")
             },
-            valid => valid
+            valid => {
+              Logger("application").error(s"AAAAAAAAAAAAAAAAAAAAA Response: $valid\n")
```
Comment (tharunprasath) [inline app/connectors/IncomeTaxCalculationConnector.scala:52]: Could you please add a valid logger?

File: app/controllers/TaxDueSummaryController.scala [hunk 2]
```diff
@@ -63,6 +63,7 @@ class TaxDueSummaryController @Inject()(val authorisedFunctions: AuthorisedFunct
     calculationService.getLiabilityCalculationDetail(user.mtditid, user.nino, taxYear).map {
       case liabilityCalc: LiabilityCalculationResponse =>
         val viewModel = TaxDueSummaryViewModel(liabilityCalc)
+
```
Comment (tharunprasath) [inline app/controllers/TaxDueSummaryController.scala:66]: Could be made unchanged

File: app/views/TaxCalcBreakdown.scala.html [hunk 3]
```diff
@@ -35,7 +35,7 @@
     <tr class="govuk-table__row">
         <td class="govuk-table__cell">
         @messages(
-            if(scottishRegime && band.name == "ART") {
+            if(scottishRegime && (band.name == "ART" || band.name == "AVRT" && taxYear >= 2024)) {
```
Comment (tharunprasath) [inline app/views/TaxCalcBreakdown.scala.html:38]: Would this be more readable like "if (scottishRegime && (band.name == "ART" || (band.name == "AVRT" && taxYear >= 2024)))". Also, can the magic number be removed with a valid name to understand the criteria like MIN_YEAR_FOR_SCOTTISH_REGIME or a better one?

File: conf/messages [hunk 4]
```diff
@@ -475,6 +475,7 @@ taxCal_breakdown.table.SRT                                          = Starter ra
 taxCal_breakdown.table.BRT                                          = Basic rate ({0} at {1}%)
 taxCal_breakdown.table.IRT                                          = Intermediate rate ({0} at {1}%)
 taxCal_breakdown.table.HRT                                          = Higher rate ({0} at {1}%)
+taxCal_breakdown.table.AVRT_scottish                                = Advanced rate ({0} at {1}%)
```
Comment (tharunprasath) [inline conf/messages:478]: Do we need to handle Welsh?

File: test/testConstants/NewCalcBreakdownUnitTestConstants.scala [hunk 5]
```diff
@@ -929,6 +929,14 @@ object NewCalcBreakdownUnitTestConstants {
         income = 100000,
         taxAmount = 40000.00
       ),
+      TaxBands(
```
Comment (tharunprasath) [inline test/testConstants/NewCalcBreakdownUnitTestConstants.scala:932]: Good to see the test for this AC "Please adhere to existing sorting method
@for(band <- viewModel.lumpSumsBands.getOrElse(Seq()).filter(.income > 0).sortBy(.rate)) { @taxBandTableRow(band, viewModel.taxRegime.contains("Scotland")) }" in both Pay, pensions and profit and Employment lump sums

**Review by tharunprasath — CHANGES_REQUESTED** [review]

**Review by tharunprasath — APPROVED** [review]

**Review by vinay1yv — APPROVED** [review]

**Pull request merged**